src/fn/create-users.js

```diff
@@ -1,52 +1,92 @@
-      const usernameIndex = cols.indexOf('username');
-      const passwordIndex = cols.indexOf('password');
-      const rolesIndex = cols.indexOf('roles');
-      const placeIdIndex = cols.indexOf('place');
-      const contactIndex = cols.indexOf('contact');
-
-      return rows.reduce((promiseChain, row) => {
-        const username = row[usernameIndex];
-        const password = row[passwordIndex];
-        const roles     = row[rolesIndex].split(':');
-        const contact = contactIndex === -1 ? prefixedProperties(cols, row, 'contact.') : row[contactIndex];
-        const place = placeIdIndex === -1 ? prefixedProperties(cols, row, 'place.') : row[placeIdIndex];
-        const requestObject = { username, password, roles, place, contact };
-
-        return promiseChain
-          .then(() => {
-            info('Creating user', username);
-            return request({
-              uri: `${instanceUrl}/api/v1/users`,
-              method: 'POST',
-              json: true,
-              body: requestObject,
-            });
-          });
-      }, Promise.resolve());
+const nestPrefixedProperties = (obj, name) => {
+  const nested = {};
+  const prefix = `${name}.`;
+  Object
+    .keys(obj)
+    .filter(key => key.startsWith(prefix))
+    .forEach(key => {
+      nested[key.substring(prefix.length)] = obj[key];
+      delete obj[key];
     });
+  return obj[name] || nested;
 };
 
-function prefixedProperties (cols, row, prefix) {
-  const indices = {};
-  cols.forEach(col => {
-    if (col.startsWith(prefix)) {
-      indices[col.substring(prefix.length)] = row[cols.indexOf(col)];
-    }
+const getUsersData = (csvData) => {
+  const users = csvParse(csvData, { columns: true });
+  users.forEach(user => {
+    user.contact = nestPrefixedProperties(user, 'contact');
+    user.place = nestPrefixedProperties(user, 'place');
+    user.roles = user.roles && user.roles.split(':');
   });
-  return indices;
-}
+  return users;
+};
+
+const getUserInfo = async (instanceUrl, user) => {
+  const getId = (obj) => typeof obj === 'string' ? obj : obj._id;
+  const facilityId = getId(user.place);
+  if (!facilityId) {
+    // new place - nothing to check
+    return;
+  }
+
+  const params = {
+    facility_id: facilityId,
+    role: JSON.stringify(user.roles),
+    contact: getId(user.contact),
+  };
+
+  info(`Requesting user-info for "${user.username}"`);
+  let result;
+  try {
+    result = await request.get(`${instanceUrl}/api/v1/users-info`, { qs: params, json: true });
+  } catch (err) {
+    // we can safely ignore some errors
+    // - 404: This endpoint was only added in 3.7
+    // - 400: The endpoint throws an error if the requested roles are "online"
+    // - 400: Missing facility or role, the corresponding user create request will fail
+    if (err.statusCode !== 404 && err.statusCode !== 400) {
+      throw err;
+    }
+  }
+  return result;
+};
+
+const createUser = async (instanceUrl, user) => {
```

Comment:
`async` is not required here

Reply:
yea, I must have rewritten the function at some point and forgot the tag. Sorry